docs/examples/ros2_compatibility.rst

```diff
@@ -0,0 +1,64 @@
+.. _examples_ros2_compatibility:
+
+ROS 2 Compatibility
+===================
+
+As stated in the :ref:`linux_ros2_compilation` section, eProsima Shapes Demo can be built and used on a ROS 2 installation with additional ROS 2 features.
+
+With ROS 2 features enabled, an additional "Use ROS2 Topics" checkbox in the Participant configuration dialog will be shown.
+
+.. image:: /01-figures/participant_ros_enabled.png
+   :scale: 100 %
+   :alt: ROS 2 Participant Configuration Options
+   :align: center
+
+Enabling this box will put Shapes Demo in ROS 2 mode.
+ROS 2 Shapes Demo main window changes to reflect this by showing a different title and the `Vulcanexus <https://www.vulcanexus.org/>`_ logo.
+
+.. image:: /01-figures/ros2_shapes_demo.png
+   :scale: 100 %
+   :alt: ROS 2 Shapes Demo Main Window
+   :align: center
+
+When using eProsima Shapes Demo in this mode, ROS 2 will be aware of the Topics transmitted.
+For instance, this is the output of the ``ros2 topic list`` command.
+
+   .. code-block:: bash
+
+      $ ros2 topic list -t
+      /Square [shapes_demo_typesupport/idl/KeylessShapeType]
+      /parameter_events [rcl_interfaces/msg/ParameterEvent]
+      /rosout [rcl_interfaces/msg/Log]
```

Comment:
This is not working for me. I'd split this in two to have an easy-to-copy command and the output on different boxes

Reply:
I believe this is a side-effect of the SHM issue between Fast DDS versions. I've reproduced the issue with the v2.9.1 (as it is on the repos file) and it's working properly when using `master` for both Shapes Demo and Fast DDS.

Reply:
You're correct! I disabled SHM and datasharing and it works